include/nbl/video/utilities/IUtilities.h

```diff
@@ -26,6 +26,29 @@ class NBL_API IUtilities : public core::IReferenceCounted
         {
             auto physicalDevice = m_device->getPhysicalDevice();
             const auto& limits = physicalDevice->getLimits();
+
+            auto queueFamProps = physicalDevice->getQueueFamilyProperties();
+            uint32_t minImageTransferGranularityVolume = 1u; // minImageTransferGranularity.width * height * depth
+
+            for (uint32_t i = 0; i < queueFamProps.size(); i++)
+            {
+                uint32_t volume = queueFamProps[i].minImageTransferGranularity.width * queueFamProps[i].minImageTransferGranularity.height * queueFamProps[i].minImageTransferGranularity.depth;
+                if(minImageTransferGranularityVolume < volume)
+                    minImageTransferGranularityVolume = volume;
+            }
+
+            const uint32_t bufferOptimalTransferAtom = limits.maxResidentInvocations*sizeof(uint32_t);
+            const uint32_t imageOptimalTransferAtom = limits.maxResidentInvocations * asset::TexelBlockInfo(asset::EF_R64G64B64A64_SFLOAT).getBlockByteSize() * minImageTransferGranularityVolume;
+            const uint32_t optimalTransferAtom = core::max(bufferOptimalTransferAtom, imageOptimalTransferAtom);
```

Comment:
you need a `minOptimalTransferAtom` and `maxOptimalTransferAtom` (a range)

you currently only get the `maxOptimalTransferAtom` (by maxing the buffer one and the largest granularity image one)

you also need the one thats a `core::min(bufferOptimalTransferAtom, minImageOptimalTransferAtom)`


P.S. `minImageOptimalTransferAtom = limits.maxResidentInvocations * asset::TexelBlockInfo(asset::EF_R8_UINT).getBlockByteSize();`